go.mod

```diff
@@ -88,7 +88,7 @@ require (
 	github.com/dimchansky/utfbom v1.1.1 // indirect
 	github.com/docker/cli v20.10.21+incompatible // indirect
 	github.com/docker/distribution v2.8.2+incompatible // indirect
-	github.com/docker/docker v20.10.24+incompatible // indirect
+	github.com/docker/docker v24.0.5+incompatible // indirect
```

Comment:
looks like this might be causing issues?

> /home/prow/go/pkg/mod/oras.land/oras-go@v1.2.2/pkg/auth/docker/login.go:86:39: cannot use remote (variable of type *"github.com/docker/docker/registry".Service) as "github.com/docker/docker/registry".Service value in argument to c.loginWithTLS
/home/prow/go/pkg/mod/oras.land/oras-go@v1.2.2/pkg/auth/docker/login_tls.go:181:30: assignment mismatch: 3 variables but registry.PingV2Registry returns 2 values

Reply:
Yes, there's an incompatibility between newer Docker packages and Helm. CAPI worked around it by not importing Helm since they were only using some constants from the package, but we don't have that option. I've been trying a variety of approaches, but no luck yet. This is also blocking v1.5.0 being integrated.

Reply:
I confirmed that removing our Helm imports would work around this. That's not a trivial amount of work, however: it involves rewriting `InstallHelmChart` to shell out to the `helm` CLI, which gives up a lot of control.

I'll keep investigating this path since I don't know of other options.

Reply:
Would switching the Helm install to CAAPH instead fix the issue? If we're installing it with clusterctl, then I believe we could bypass the need to import Helm directly.

Reply:
That's definitely the way to go, but since we need this change for CAPI v1.4.5 I think we should merge this first, then apply the CAAPH PR afterward.